### PR TITLE
[OSD-16025] Add scrape interval parameter to the prometheus config

### PR DESCRIPTION
**Pull request by zmird-r**
- Needed for improving performance on hypershift management clusters

Ticket: [Increase OBO global scrape interval from 30s to 60s](https://issues.redhat.com/browse/OSD-16025)
Epic: [OBO Performance Improvements](https://issues.redhat.com/browse/SDE-2823)

### CODE REVIEW

File: pkg/apis/monitoring/v1alpha1/types.go [hunk 1]
```diff
@@ -184,6 +184,10 @@ type PrometheusConfig struct {
 	// Enable Prometheus to be used as a receiver for the Prometheus remote write protocol. Defaults to the value of `false`.
 	// +optional
 	EnableRemoteWriteReceiver bool `json:"enableRemoteWriteReceiver,omitempty"`
+    // Scrape interval for prometheus.
```
Comment (simonpasquier) [inline pkg/apis/monitoring/v1alpha1/types.go:187]:
```suggestion
    // Default interval between scrapes.
```

File: pkg/controllers/monitoring/monitoring-stack/components.go [hunk 2]
```diff
@@ -172,6 +172,7 @@ func newPrometheus(
 				RemoteWrite:               config.RemoteWrite,
 				ExternalLabels:            config.ExternalLabels,
 				EnableRemoteWriteReceiver: config.EnableRemoteWriteReceiver,
+				ScrapeInterval:            config.ScrapeInterval,
```
Comment (simonpasquier) [inline pkg/controllers/monitoring/monitoring-stack/components.go:175]: If we make it a pointer then we need to test for nil.

Comment (zmird-r) [reply pkg/controllers/monitoring/monitoring-stack/components.go]: Added a check for nil :)

**Comment by simonpasquier** [comment]
we have special requirements about commit message format :) 
https://github.com/rhobs/observability-operator/blob/main/docs/developer.md#commit-message-convention

You need also to run `go fmt ./...` and `make generate`.

**Comment by zmird-r** [comment]
Thank you for the suggestions Simon. I merged all the commits into one with the correct message format.

**Review by sthaha — APPROVED** [review]
Looks great! 
Thank you for the patch @zmird-r
